### PR TITLE
UI: fix redirect after creating new version of a KV v2 secret in a namespace

### PR DESCRIPTION
**Pull request by hellobontempo**
This fixes an issue in a namespace where the UI was not navigating the latest secret version details view after creating a new secret version.

This was also tested in a namespace with a policy that had `CREATE` but no `READ` capabilities

```shell
# sample policy
path "test-ns-1/kv-ns/data/*" {
  capabilities = [ "create", "update", "list" ]
}

path "test-ns-1/kv-ns/metadata/*" {
  capabilities = [ "create", "update", "list"]
}
```

Add-on to work in this PR: https://github.com/hashicorp/vault/pull/22426

### CODE REVIEW

File: ui/lib/kv/addon/components/page/secret/edit.js [hunk 1]
```diff
@@ -58,10 +58,11 @@ export default class KvSecretEdit extends Component {
       this.invalidFormAlert = invalidFormMessage;
       if (isValid) {
         const { secret } = this.args;
-        yield this.args.secret.save();
+        yield secret.save();
         this.flashMessages.success(`Successfully created new version of ${secret.path}.`);
-        // transition to parent secret route to re-query latest version
-        this.router.transitionTo('vault.cluster.secrets.backend.kv.secret');
+        this.router.transitionTo('vault.cluster.secrets.backend.kv.secret.details', secret.path, {
```
Comment (hashishaw) [inline ui/lib/kv/addon/components/page/secret/edit.js:63]: if we do include `secret.path` on the transition we'll need to do some encoding and test with nested secret paths as well. If it was working reliably without, it's probably easiest to remove

File: ui/tests/acceptance/secrets/backend/kv/kv-v2-workflow-edge-cases-test.js [hunk 2]
```diff
@@ -242,3 +242,88 @@ module('Acceptance | kv-v2 workflow | edge cases', function (hooks) {
...
+
+  module('admin persona', function (hooks) {
+    hooks.beforeEach(async function () {
+      await authPage.loginNs(this.namespace);
+      // mount engine within namespace
+      await runCmd(mountEngineCmd('kv-v2', this.backend), false);
+      return;
+    });
+    hooks.afterEach(async function () {
+      // visit logout with namespace query param because we're transitioning from within an engine
```
Comment (hashishaw) [inline ui/tests/acceptance/secrets/backend/kv/kv-v2-workflow-edge-cases-test.js:278]: 😭 for issue 👏 for comment

File: ui/lib/kv/addon/components/page/secret/edit.js (hunk 1 above)
Comment (hellobontempo) [reply ui/lib/kv/addon/components/page/secret/edit.js:63]: Yeah - I forgot we had added to try and resolve test issues. Will remove!

File: ui/tests/acceptance/secrets/backend/kv/kv-v2-workflow-edge-cases-test.js (hunk 2 above)
Comment (hellobontempo) [reply ui/tests/acceptance/secrets/backend/kv/kv-v2-workflow-edge-cases-test.js:278]: 😅

**Review by hashishaw — APPROVED** [review]
🙏
